test/test_load_balancer.py

```diff
@@ -45,7 +45,7 @@ def test_connections_are_only_returned_once(self):
         nconns = len(pool.sockets)
         self.db.test.find_one({})
         self.assertEqual(len(pool.sockets), nconns)
-        self.db.test.aggregate([{'$limit': 1}])
+        list(self.db.test.aggregate([{'$limit': 1}]))
```

Comment:
I suspect the problem here was that we didn't explicitly cleanup or exhaust the cursor. So sometimes the connection would remain pinned to the cursor until it eventually gets garbage collected at some later point.